pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -303,7 +314,7 @@ public void run() {
     });
 
     indexingThread.start();
-
+    serverMetrics.addValueToTableGauge(tableName, ServerGauge.SEGMENT_COUNT, 1L);
```

Comment:
There are two scenarios we will hit realtime segment shutting down:
1. server shutdown
2. realtime table deletion.
I think in both cases, we should be fine.